Restructure docs (#285)
elastic/crawler: docs/DEVELOPER_GUIDE.md

1. Clone the repository: `git clone https://github.com/elastic/crawler.git`
2. Go to the root of the Open Crawler directory and check the expected Java and Ruby versions are being used:
    ```bash
    # should output the same version as `.ruby-version`
    ruby --version

    # should output the same version as `.java-version`
    java -version
    ```
3. If the versions seem correct, you can install dependencies:
    ```bash
    make install
    ```

   You can also use the env variable `CRAWLER_MANAGE_ENV` to have the install script automatically check whether `rbenv` and `jenv` are installed, and that the correct versions are running on both:
   Doing this requires that you use both `rbenv` and `jenv` in your local setup.

    ```bash
    CRAWLER_MANAGE_ENV=true make install
    ```
4. Now you should be able to run Crawler locally
    ```bash
   bin/crawler crawl path/to/config.yml
   ```

## Configuration

The crawler details need to be provided in a crawler configuration file.
You can specify Elasticsearch instance configuration within that file, or optionally in a separate configuration file.

> This is possible so that multiple crawlers can run using a single Elasticsearch configuration.

Reviewer: ```suggestion
This allows multiple crawlers to share a single Elasticsearch configuration.
```
